src/Lecture3.hs

```diff
@@ -210,8 +246,23 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+
+instance Foldable List1 where
+  foldMap f (List1 x xs)  = f x `mappend` foldMap f xs
+
+  foldr f z (List1 x xs) = f x (foldr f z xs)
+
+
+instance Foldable Treasure where
+  foldMap = helper mempty
+    where
+      helper :: p -> (t -> p) -> Treasure t -> p
+      helper x _ NoTreasure = x
+      helper _ f (SomeTreasure x) = f x
```

Comment:
I think you don't need a `helper` function here and can implement the same logic on the top-level for `foldMap` directly

Reply:
yup